csharp/src/Ice/StreamWrapper.cs

```diff
@@ -5,86 +5,72 @@
 using System.Diagnostics;
 using System.IO;
 
-namespace IceInternal
+namespace Ice
 {
-    //
     // Classes to provide a System.IO.Stream interface on top of an Ice stream.
-    // We use this to serialize arbitrary .NET serializable classes into
-    // a Slice byte sequence.
     //
-    // For input streams, this class is a wrapper around the InputStream
-    // class that passes all methods through.
+    // We use this to serialize arbitrary .NET serializable classes into a Slice byte sequence.
+    //
+    // For input streams, this class is a wrapper around the InputStream class that passes all methods through.
     //
     // For output streams, we use a different strategy:
-    // Slice sequences are encoded on the wire as a count of elements, followed
-    // by the sequence contents. For arbitrary .NET classes, we do not know how
-    // big the sequence that is eventually written will be. To avoid excessive
-    // data copying, this class maintains a private _bytes array of 254 bytes and,
-    // initially, writes data into that array. If more than 254 bytes end up being
-    // written, we write a dummy sequence size of 255 (which occupies five bytes
-    // on the wire) into the stream and, once this class is disposed, patch
-    // that size to match the actual size. Otherwise, if the _bytes buffer contains
-    // fewer than 255 bytes when this class is disposed, we write the sequence size
-    // as a single byte, followed by the contents of the _bytes buffer.
     //
-
-    public class OutputStreamWrapper : Stream
+    // Slice sequences are encoded on the wire as a count of elements, followed by the sequence contents. For arbitrary
+    // .NET classes, we do not know how big the sequence will be. To avoid excessive data copying, this class maintains
+    // a private _bytes array of 254 bytes and, initially, writes data into that array. If more than 254 bytes end up
+    // being written, we write a dummy sequence size of 255 (which occupies five bytes on the wire) into the stream and,
+    // once this class is disposed, patch that size to match the actual size. Otherwise, if the _bytes buffer contains
+    // fewer than 255 bytes when this class is disposed, we write the sequence size as a single byte, followed by the
+    // contents of the _bytes buffer.
+    internal class OutputStreamWrapper : Stream
     {
-        public OutputStreamWrapper(Ice.OutputStream s)
+        private readonly OutputStream _stream;
+        private OutputStream.Position _startPos;
+        private byte[]? _data;
+        private int _pos;
+        private long _length;
```

Comment:
Probably don't need _length.

Reply:
We use _length for `Length/SetLength` overrides

Reply:
It should just throw NotSupported exception since this stream wrapper does not support seeking:
https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.length?view=net-5.0